tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -1342,6 +1372,37 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
         return S_FALSE; // TODO https://datadoghq.atlassian.net/browse/DEBUG-706
     }
 
+    if (!isAsyncMethod)
+    {
+        // Async methods can't have Task<T> where T is byref like, because byref like can't exist as a generic param.
+        // thus we only perform this check here.
```

Comment:
```suggestion
        // In async methods, the return value can't be byref-like (it can't be Task<T> where T is byref-like, because byref-like can't exist as a generic param).
        // There, we only need to worry about non-async methods.
```

Reply:
Great, thanks!